TUnit.Assertions/Assertions/Generics/DoesExtensions_Generic.cs

```diff
@@ -16,10 +16,10 @@ public static InvokableValueAssertionBuilder<TActual> Contains<TActual, TInner>(
             , [doNotPopulateThisValue]);
     }
     
-    public static InvokableValueAssertionBuilder<IEnumerable<TInner>> Contains<TInner>(this IValueSource<IEnumerable<TInner>> valueSource, Func<TInner, bool> matcher, [CallerArgumentExpression(nameof(matcher))] string doNotPopulateThisValue = null)
+    public static MappableResultAssertionBuilder<IEnumerable<TInner>, TInner> Contains<TInner>(this IValueSource<IEnumerable<TInner>> valueSource, Func<TInner, bool> matcher, [CallerArgumentExpression(nameof(matcher))] string doNotPopulateThisValue = null)
     {
-        return valueSource.RegisterAssertion(new EnumerableContainsExpectedFuncAssertCondition<IEnumerable<TInner>, TInner>(matcher, doNotPopulateThisValue)
-            , [doNotPopulateThisValue]);
+        return new MappableResultAssertionBuilder<IEnumerable<TInner>, TInner>(valueSource.RegisterAssertion(new EnumerableContainsExpectedFuncAssertCondition<IEnumerable<TInner>, TInner>(matcher, doNotPopulateThisValue)
```

Comment:
This will enumerate the `IEnumerable` twice and also evaluate `matcher` for every item twice, right?

Reply:
If awaited directly yeah. Didn't think about that. Maybe I can opt to pull it from the assert condition instead.